src/types/intDecoding.d.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Configure how integers in JSON response will be decoded.
+ *
+ *   The options are:
+ *   * "default": All integers will be decoded as Numbers, meaning any values greater than
+ *     Number.MAX_SAFE_INTEGER will lose precision.
+ *   * "safe": All integers will be decoded as Numbers, but if any values are greater than
+ *     Number.MAX_SAFE_INTEGER an error will be thrown.
+ *   * "mixed": Integers will be decoded as Numbers if they are less than or equal to
+ *     Number.MAX_SAFE_INTEGER, otherwise they will be decoded as BigInts.
+ *   * "bigint": All integers will be decoded as BigInts.
+ *
+ *   Defaults to "default" if not included.
+ */
+export type IntDecoding = 'default' | 'safe' | 'mixed' | 'bigint';
```

Comment:
I think an enum might make more sense for this field, let me know if you disagree.

Reply:
Yes, definitely. Initially I wasn't sure whether a union was needed for implementation reasons. I'll convert it to an enum.